testplan/web_ui/testing/src/AssertionPane/AssertionTypes/GraphAssertions/XYGraphAssertion.js

```diff
@@ -0,0 +1,138 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import 'react-vis/dist/style.css';
+import './App.css';
+import * as GraphUtil from './graphUtils';
+
+import {
+  XAxis,
+  YAxis,
+  DiscreteColorLegend,
+  HorizontalGridLines,
+  XYPlot,
+  LineSeries,
+  VerticalBarSeries,
+  Highlight,
+  HexbinSeries,
+  ContourSeries,
+  WhiskerSeries
+} from 'react-vis';
+
+
+/**
+ * Component that are used to render a Graph (Data visualisations that require an XY axis).
+ */
+class XYGraphAssertion extends Component {
+    constructor(props) {
+      super(props);
+
+      this.series_colour = {}
+      let data = this.props.assertion.graph_data;
+      let series_options = this.props.assertion.series_options;
+
+      var plot_options;
+      for (var key in data) {
+           if(series_options !== null){
+               plot_options = series_options[key];
+            }
+          var plot_colour = GraphUtil.returnColour(plot_options);
+          this.series_colour[key] = plot_colour;
+      }
+    }
+
+
+    state = {
+        lastDrawLocation: null
+     };
+
+
+    components = {
+        Line: LineSeries,
+        Hexbin: HexbinSeries,
+        Contour: ContourSeries,
+        Whisker: WhiskerSeries,
+        Bar: VerticalBarSeries
+    }
+
+
+  render() {
+    let data = this.props.assertion.graph_data;
+   // let series_options = this.props.assertion.series_options;
```

Comment:
What is happening with series_options?

Reply:
Here series options are used to setup the colours of the plots in the constructor (to avoid the disco colours every time the component re-renders because of react), but apart from that it is not used in the render() method as colour is the only available series option at the moment